doc/nrf/ug_multi_image.rst

```diff
@@ -24,12 +42,134 @@ MCUboot bootloader
+To ignore an MCUboot child image, select :option:`CONFIG_MCUBOOT_BUILD_STRATEGY_SKIP_BUILD` instead of :option:`CONFIG_MCUBOOT_BUILD_STRATEGY_FROM_SOURCE`.
+
+
+Defining and enabling a child image
+***********************************
+
+You can enable existing child images in the |NCS| by enabling the respective modules in the parent image and selecting the desired build strategy. (? is that correct?)
+To turn an application that you have implemented into a child image that can be included in a parent image, you must add the required configuration options, update the build scripts to make it possible to enable the child image, and make sure that image-specific variables can be disambiguated.
+
+Adding configuration options
+============================
+
+When enabling a child image, you select the *build strategy*, thus how the image is included.
+The three options are:
+
+* Build the child image from source along with the parent image- *IMAGENAME*\_BUILD_STRATEGY_FROM_SOURCE
+* Merge the specified HEX file of the child image with the parent image - *IMAGENAME*\_BUILD_STRATEGY_USE_HEX_FILE, and *IMAGENAME*\_HEX_FILE to specify the HEX file
+* Ignore the child image when building and build only the parent image - *IMAGENAME*\_BUILD_STRATEGY_SKIP_BUILD
+
+You must add these four configuration options to the Kconfig file for your child image, replacing *IMAGENAME* with the (uppercase) name of your child image.
+(? are there any rules for the name? Does it have to be the same as the file name, for example?)
+
+The following example shows the configuration options for MCUboot:
+(? this is mentioning zephyr_add_executable, which doesn't exist anymore?)
+
+.. code-block:: Kconfig
+
+   choice
+  	prompt "MCUboot build strategy"
+  	default MCUBOOT_BUILD_STRATEGY_FROM_SOURCE
+
+   config MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE
+  	# Mandatory option when being built through 'zephyr_add_executable'
+  	bool "Use HEX file instead of building MCUboot"
+
+   if MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE
+
+   config MCUBOOT_HEX_FILE
+  	# Mandatory option when being built through 'zephyr_add_executable'
+  	string "MCUboot HEX file"
+
+   endif # MCUBOOT_USE_HEX_FILE
+
+   config MCUBOOT_BUILD_STRATEGY_SKIP_BUILD
+  	# Mandatory option when being built through 'zephyr_add_executable'
+  	bool "Skip building MCUboot"
+
+   config MCUBOOT_BUILD_STRATEGY_FROM_SOURCE
+  	# Mandatory option when being built through 'zephyr_add_executable'
+  	bool "Build from source"
+
+   endchoice
+
+
+Updating the build scripts
+==========================
+
+To make it possible to enable a child image from a parent image, you must include the child image in the build script.
+This code should be put in place in the cmake tree that is conditional on a configuration option for having the parent image use the child image.
+
+In the |NCS|, the code is included in the :file:`CMakeLists.txt` file for the samples. (? where is the mcuboot code?)
```

Comment:
> (? where is the mcuboot code?)


Can you be more specific?

Reply:
In samples/CMakeLists.txt, I see code like the one mentioned for SPM and for SECURE_BOOT. But not for mcuboot - where is that located?

Reply:
It's here:

https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/blob/master/zephyr/CMakeLists.txt#L86